deltachat-ios/Controller/Settings/Proxy/ProxySettingsViewController.swift

```diff
@@ -0,0 +1,285 @@
+
+                let proxyUrl = proxies[indexPath.row]
+
+                let connectionStateText: String?
+
+                if let selectedProxy, selectedProxy == proxyUrl {
+                    cell.accessoryType = .checkmark
+                    if dcContext.isProxyEnabled {
+                        connectionStateText = DcUtils.getConnectivityString(dcContext: dcContext, connectedString: String.localized("connectivity_connected"))
+                    } else {
+                        connectionStateText = String.localized("connectivity_not_connected")
+                    }
+                } else {
+                    cell.accessoryType = .none
+                    connectionStateText = nil
+                }
+
+                cell.configure(with: proxyUrl, dcContext: dcContext, connectionStateText: connectionStateText)
+
+                return cell
+            } else /*if indexPath.section == ProxySettingsSection.add.rawValue*/ {
+                return addProxyCell
+            }
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        guard proxies.isEmpty == false else { return nil }
+
+        if section == ProxySettingsSection.proxies.rawValue {
+            return ProxySettingsSection.proxies.title
+        } else {
+            return nil
+        }
+
+    }
+}
+
+// MARK: - UITableViewDelegate
+
+extension ProxySettingsViewController {
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if proxies.isEmpty {
+            if indexPath.section == ProxySettingsSection.enableProxies.rawValue {
+                // do nothing as there are no proxies that could be used
+            } else /* if indexPath.section == ProxySettingsSection.add.rawValue */ {
+                addProxy()
+            }
+        } else {
+            if indexPath.section == ProxySettingsSection.enableProxies.rawValue {
+                toggleProxyCell.uiSwitch.isOn.toggle()
+                toggleProxyCell.didToggleSwitch(toggleProxyCell.uiSwitch)
+            } else if indexPath.section == ProxySettingsSection.proxies.rawValue {
+                selectProxy(at: indexPath)
+            } else /*if indexPath.section == ProxySettingsSection.add.rawValue*/ {
+                addProxy()
+            }
+        }
+
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+
+    override func tableView(_ tableView: UITableView, trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+        guard
+            proxies.isEmpty == false,
+            indexPath.section == ProxySettingsSection.proxies.rawValue
+        else { return nil }
+
+        let deleteAction = UIContextualAction(style: .destructive, title: String.localized("proxy_delete")) { [weak self] _, _, completion in
```

Comment:
context is clear here:

```suggestion
        let deleteAction = UIContextualAction(style: .destructive, title: String.localized("delete")) { [weak self] _, _, completion in
```